libbpf-tools/tcppktlat.c

```diff
@@ -171,6 +230,134 @@ static void handle_lost_events(void *ctx, int cpu, __u64 lost_cnt)
 	fprintf(stderr, "lost %llu events on CPU #%d\n", lost_cnt, cpu);
 }
 
+static void calc_hist_stats(__u32 *slots, int slots_size, double *min,
+			    double *max, double *avg, double *mean, double *p95,
+			    double *p99)
+{
+	unsigned long long total = 0;
+	unsigned long long sum = 0;
+	int i, min_slot = -1, max_slot = -1;
+	unsigned long long low, high, mid;
+
+	/* Calculate total count and find min/max slots */
+	for (i = 0; i < slots_size; i++) {
+		if (slots[i] > 0) {
+			if (min_slot < 0)
+				min_slot = i;
+			max_slot = i;
+			total += slots[i];
+		}
+	}
+
+	if (min_slot < 0) {
+		*min = *max = *avg = *mean = *p95 = *p99 = 0;
+		return;
+	}
+
+	/* Calculate min (low bound of first slot) */
+	low = (1ULL << (min_slot + 1)) >> 1;
+	if (low == (1ULL << (min_slot + 1)) - 1)
+		low -= 1;
+	*min = low;
+
+	/* Calculate max (high bound of last slot) */
+	high = (1ULL << (max_slot + 1)) - 1;
+	*max = high;
+
+	/* Calculate avg (average) */
+	for (i = 0; i < slots_size; i++) {
+		if (slots[i] > 0) {
+			low = (1ULL << (i + 1)) >> 1;
+			high = (1ULL << (i + 1)) - 1;
+			if (low == high)
+				low -= 1;
+			/* Use midpoint of the range as representative value */
+			mid = (low + high) / 2;
+			sum += mid * slots[i];
+		}
+	}
+	*avg = total > 0 ? (double)sum / total : 0;
+
+	/* Calculate mean (p50, median) and percentiles */
+	unsigned long long p50_count = total * 50 / 100;
+	unsigned long long p95_count = total * 95 / 100;
+	unsigned long long p99_count = total * 99 / 100;
+	unsigned long long cumsum = 0;
+
+	*mean = *p95 = *p99 = 0;
+	for (i = 0; i < slots_size; i++) {
+		if (slots[i] > 0) {
+			cumsum += slots[i];
+			low = (1ULL << (i + 1)) >> 1;
+			high = (1ULL << (i + 1)) - 1;
+			if (low == high)
+				low -= 1;
+
+			if (*mean == 0 && cumsum >= p50_count) {
+				*mean = high; /* Use high bound for median */
+			}
+			if (*p95 == 0 && cumsum >= p95_count) {
+				*p95 = high; /* Use high bound for percentile */
+			}
+			if (*p99 == 0 && cumsum >= p99_count) {
+				*p99 = high; /* Use high bound for percentile */
+			}
+		}
+	}
+}
+
+static int print_hist(struct bpf_map *hists_map)
```

Comment:
When running the command:

$ sudo ./tcppktlat -H 5 3

I noticed that during the 3 outputs, one output showed data for 0 PIDs, one showed 2 PIDs, and one showed 1 PID. However, it's unclear which time interval each output corresponds to.

Would it make sense to add something like a timestamp for each output to indicate the time period of the reported data? This would help users better understand and analyze the latency data over time.